intake_pcap/stream.py

```diff
@@ -82,7 +83,15 @@ def decoder(header, data):
 
         columns = FullPacket._fields if self._payload else BasePacket._fields
         df = pd.DataFrame(packets, columns=columns)
-        return df.astype(dtype=self.dtype)
+
+        # DataFrame.astype doesn't work with extension types (yet).
+        # https://github.com/pandas-dev/pandas/issues/20557
+        known_types = {k: v for k, v in self.dtype.items()
+                       if k not in ('src_host', 'dst_host')}
+        df = df.astype(known_types)
+        df['src_host'] = to_ipaddress(df['src_host'])
```

Comment:
What is the input `df['src_host']` here? Are we parsing text, bytes or just copying binary data? Seems like there ought to be a way of making this not-slow.

Reply:
AFAICT, at this point `src_host` and `dst_host` are strings.

Reply:
One issue is that the fastest way to build an IPArray is from a columar 64-bit aligned bytestring, whereas these seem to be record based. I'll look a bit deeper (after profiling things).

Reply:
If the input is not already an array of the right structure (e.g., with `.view()`), it may be useful here to build the new columns using empty arrays and fill them, perhaps in a tight numba loop. 

That will be generally true for the rest of the dataframe too.